TFT/src/User/API/Settings.c

```diff
@@ -214,34 +221,29 @@ void setupMachine(void)
   #endif
 
   if (infoMachineSettings.leveling != BL_DISABLED && infoMachineSettings.EEPROM == 1 && infoSettings.auto_load_leveling == 1)
-  {
     storeCmd("M420 S1\n");
-  }
 
-  if (infoMachineSettings.firmwareType != FW_MARLIN)  // Smoothieware does not report detailed M115 capabilities
-  {
-    infoMachineSettings.EEPROM                  = ENABLED;
-    infoMachineSettings.autoReportTemp          = DISABLED;
-    infoMachineSettings.autoReportPos           = DISABLED;
-    infoMachineSettings.leveling                = ENABLED;
-    infoMachineSettings.zProbe                  = ENABLED;
-    infoMachineSettings.levelingData            = ENABLED;
-    infoMachineSettings.emergencyParser         = ENABLED;
-    infoMachineSettings.autoReportSDStatus      = DISABLED;
-  }
   if (infoSettings.onboard_sd != AUTO)
     infoMachineSettings.onboardSD = infoSettings.onboard_sd;
 
   if (infoSettings.long_filename != AUTO)
     infoMachineSettings.longFilename = infoSettings.long_filename;
 
-  mustStoreCmd("M503 S0\n");
+  if (infoMachineSettings.firmwareType == FW_SMOOTHIEWARE)  // Smoothieware does not report detailed M115 capabilities
+  {
+    infoMachineSettings.leveling        = BL_ABL;
+    infoMachineSettings.emergencyParser = ENABLED;
+  }
 
   if (infoMachineSettings.firmwareType == FW_REPRAPFW)
```

Comment:
I would use "else if" for redundant "if" statements

Reply:
yes